Add new front page to the website

cdrage opened the pull request:
<!--
Thank you for opening a PR! Here are some things you need to know before submitting:

1. Please read our developer guideline: https://github.com/redhat-developer/odo/wiki/Developer-Guidelines
2. Label this PR accordingly with the '/kind' line
3. Ensure you have written and ran the appropriate tests: https://github.com/redhat-developer/odo/wiki/Writing-and-running-tests
4. Read how we approve and LGTM each PR: https://github.com/redhat-developer/odo/wiki/PR-Review

Documentation:

If you are pushing a change to documentation, please read: https://github.com/redhat-developer/odo/wiki/Contributing-to-Docs
-->

**What type of PR is this:**

<!--
Add one of the following kinds:
/kind bug
/kind feature
/kind cleanup
/kind tests
/kind documentation

Feel free to use other [labels](https://github.com/redhat-developer/odo/labels) as needed. However one of the above labels must be present or the PR will not be reviewed. This instruction is for reviewers as well.
-->

/kind documentation

**What does this PR do / why we need it:**

The website has been redesigned! Mostly the front page, but there is a
new colour scheme, logo as well as updated documentation.

Here is the summary:

- We have a new (TEMPORARY!) logo! This is not 100% official (yet) but
rather a temporary logo
- Docusaurus has added support for Typescript now and the front page is
now .tsx
- New colour scheme! It's blue similar to other k8s / container like
tools
- Two videos have been added, a container ship and a person coding to
illustrate what odo is about
- Updated links to slack (we should have it link to the slack invite web
page, rather than a direct link to the channel)

TODO:
- Eventually we will added icons to the top of the "information" blurbs
on the bottom of the page

**Which issue(s) this PR fixes:**
<!--
Specifying the issue will automatically close it when this PR is merged
-->

N/A

**PR acceptance criteria:**

- [X] Unit test

N/A

- [X] Integration test

N/A

- [X] Documentation

**How to test changes / Special notes to the reviewer:**

View the site :)

dharmit commented on docs/website/src/pages/index.tsx:
```diff
@@ -0,0 +1,135 @@
+    Standalone client
+                </h2>
+                <p className="padding-horiz--md">
+    <code>odo</code> is a standalone tool that communicates directly with the container orchestrator API. There is no requirement for a daemon or server process.
```
Why not just use Kubernetes instead? I'm suggesting this because, in my head, the more times "Kubernetes" appears on the website, the more we will be able to convey that odo is a Kubernetes tool!
```suggestion
    <code>odo</code> is a standalone tool that communicates directly with the Kubernetes API. There is no requirement for a daemon or server process.
```

dharmit commented on docs/website/src/pages/index.tsx:
```diff
@@ -0,0 +1,135 @@
+                  Built for container engineers
+                </h2>
+                <p className="padding-horiz--md">
+      Built from the ground up with application development on container orchestrators in mind. Each command has been carefuly crafted for application container development.
```
OK, I'm starting to feel that we should actually use "Kubernetes" because other container orchestrators like Hashicorp Nomad or Apache Mesos won't be a good place to try odo.
```suggestion
      Built from the ground up with application development on Kubernetes in mind. Each command has been carefuly crafted for application container development.
```

dharmit commented on docs/website/src/pages/index.tsx:
```diff
@@ -0,0 +1,135 @@
+                  Run your tests directly on the cluster
+                </h2>
+                <p className="padding-horiz--md">
+    Debug and test remote applications deployed using <code>odo</code> directly from your IDE on your container orchestrator. No more having to exit your IDE to push your application.
```
```suggestion
    Debug and test remote applications deployed using <code>odo</code> directly from your IDE on Kubernetes. No more having to exit your IDE to push your application.
```

dharmit commented on docs/website/docusaurus.config.js:
```diff
@@ -93,7 +98,7 @@ module.exports = {
         },
       ],
-      // copyright: `Copyright © ${new Date().getFullYear()} My Project, Inc. Built with Docusaurus.`,
+       copyright: `Copyright © ${new Date().getFullYear()} Odo Authors -- All Rights Reserved <br> Apache License 2.0 open source project`,
```
I don't know if we should be using "odo" vs. "Odo" in the copyright. But I might be nitpicking because it's taken us some back and forth in the past to decide upon odo vs. Odo.
```suggestion
       copyright: `Copyright © ${new Date().getFullYear()} odo Authors -- All Rights Reserved <br> Apache License 2.0 open source project`,
```

kadel replied:
long time back we decided that it is "odo", we should use it everywhere

dharmit reviewed (changes requested):
Some quick comments from looking at the homepage. TBH, this looks **AWESOME**!

cdrage commented:
Thanks @dharmit and @kadel 

I've made the following changes:
- Emphasis on Kubernetes / removal of the wording for container orchestrator
- Uploaded new icons (have the license for them) for the information blurbs
- Changed Odo to odo
- Removed red from OpenShift text + Kubernetes blue. So now it's more neutral.

Ready for another review!

dharmit commented:
/lgtm
/approve

Thanks @cdrage !
